### DISCUSSION TITLE
Issue Regarding Solving ∆P Using ROCOF

### DISCUSSION THREAD

**Discussion by lin-1l1** [CURENT/andes#537] [Q&A]
Dear Andes developers,

In the example of "Load Frequency Control," the method employed was `ss.PV.as_df()` to analyze the lost active power within the system (0.4pu). However,I want to utilize ROCOF to solve for ∆P, but I'm encountering some difficulties.

Firstly, I'm uncertain about how to choose an appropriate `Tw` value in ROCOF. Additionally, I'd like to inquire whether Andes provides a function to calculate the overall inertia H?

My calculation method is based on the ROCOF formula: `ROCOF = (∆P * f) / (2 * H * S)`, but the ∆P value I'm obtaining is only 0.2, which is not consistent with my expectations. Could I have overlooked or made a mistake somewhere in my calculations?

Thank you very much for your assistance!

Attached is the result figure I've attempted.
<img width="733" alt="截圖 2024-04-26 凌晨1 10 38" src="https://github.com/CURENT/andes/assets/167511120/80d5ebe4-5ae9-4a32-9428-5ae42a2db849">

**Comment by jinningwang** [comment]
> Firstly, I'm uncertain about how to choose an appropriate Tw value in ROCOF.

Check this, https://github.com/CURENT/andes/pull/529

> Additionally, I'd like to inquire whether Andes provides a function to calculate the overall inertia H?

Short answer, no.
For synchronous machine, H is a constant, and you can sum up `M` for all `GENCLS` and `GENROU`. However, to my knowledge, when it comes to IBR, there is no clear definition of inertia yet. You can define your own "inertia" and calculate it.

> My calculation method is based on the ROCOF formula: ROCOF = (∆P * f) / (2 * H * S), but the ∆P value I'm obtaining is only 0.2, which is not consistent with my expectations. Could I have overlooked or made a mistake somewhere in my calculations?

In this equation, delta P can be impacted by how the ROCOF is obtained, and you can pay attention to the ROCOF values first.

**Reply by lin-1l1** [reply]
jinning, thank you for your response and assistance!

After carefully reading #529, I'm still a bit confused. I've attempted to compare the results of GENROU with different `tw` values of BusROCOF, and I'm wondering if I understand correctly: Is finding the `tw` value that best approximates the smooth curve the best answer?
![image](https://github.com/CURENT/andes/assets/167511120/f46ffc58-1d39-4a96-a8e0-ccff18981717)
![image](https://github.com/CURENT/andes/assets/167511120/7899a119-e5c3-4fbd-87ca-de42e4c61f3a)

**Reply by cuihantao** [reply]
You might want to install a BusFreq at a generator bus and then tune the parameters of `BusFreq` to make the estimated frequency close to the generator's `omega`. Then, you can reuse the parameter for BusROCOF.

**Reply by jinningwang** [reply]
As shown below, `BusROCOF` is derived from `BusFreq`, so the first step is to have appropriate parameters of the bus measurement part.

https://github.com/CURENT/andes/blob/5ab784b75347a98d07c004e32d7c4b39d407ea2e/andes/models/measurement/busrocof.py#L9-L26

**Reply by lin-1l1** [reply]
Thank you Professor Cui and Jinning for your replies and assistance! Your valuable suggestions and help are greatly appreciated. Thank you once again.
